tests/validation/tests/v3_api/test_workload.py

```diff
@@ -995,83 +982,70 @@ def test_rbac_wl_daemonset_update(role, remove_resource):
         user_token = rbac_get_user_token_by_role(CLUSTER_OWNER)
         p_client = get_project_client_for_token(project, user_token)
         ns = rbac_get_namespace()
-        workload = p_client.create_workload(name=name,
-                                            containers=con,
-                                            namespaceId=ns.id,
-                                            daemonSetConfig={})
+        workload = create_workload_by_type(p_client, name, con, ns, config)
         wait_for_wl_to_active(p_client, workload)
         with pytest.raises(ApiError) as e:
             user_token = rbac_get_user_token_by_role(role)
             p_client = get_project_client_for_token(project, user_token)
             con = [{"name": "test1", "image": os.environ.get('RANCHER_TEST_IMAGE',
                                                          "nginx")}]
             p_client.update(workload, containers=con)
+            wait_for_pods_in_workload(p_client, workload)
+            validate_workload(p_client, workload, "deployment", ns.name)
         assert e.value.error.status == 403
         assert e.value.error.code == 'Forbidden'
         remove_resource(workload)
 
 
 @if_test_rbac
 @pytest.mark.parametrize("role", RBAC_ROLES)
-def test_rbac_wl_daemonset_update_negative(role, remove_resource):
+@pytest.mark.parametrize("config", WORKLOAD_TYPES)
+def test_rbac_wl_parametrize_update_negative(role, remove_resource, config):
     if role == CLUSTER_OWNER:
-        # cluster owner can edit daemonsets in any project in the cluster
-        project = rbac_get_project()
+        # workloads of type job cannot be edited
+        if config == "job":
+            return
```

Comment:
how about when the role is not cluster owner and config is job?

Reply:
Moved this check to the beginning of the method. This should be the first thing that is checked regardless of RBAC role.